tools/datagen/src/bin/verify-zero-copy.rs

```diff
@@ -0,0 +1,253 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use clap::{App, Arg, ArgGroup};
+use icu_datagen::get_all_keys;
+use icu_provider::datagen::{DataConverter, HeapStatsMarker, OmnibusDatagenProvider};
+use icu_provider::filter::Filterable;
+use icu_provider::fork::by_key::ForkByKeyProvider;
+use icu_provider::iter::IterableDynProvider;
+use icu_provider::prelude::*;
+use icu_provider::serde::SerializeMarker;
+use icu_provider_blob::BlobDataProvider;
+use icu_provider_cldr::CldrPathsAllInOne;
+use litemap::LiteMap;
+use simple_logger::SimpleLogger;
+use std::borrow::Cow;
+use std::cmp;
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::Read;
+use std::mem::ManuallyDrop;
+use std::path::PathBuf;
+use std::rc::Rc;
+
+#[global_allocator]
+static ALLOC: dhat::Alloc = dhat::Alloc;
+
+static EXPECTED_VIOLATIONS: &[&str] = &[
+    "datetime/skeletons@1",
+    "list/and@1",
+    "list/or@1",
+    "list/unit@1",
+    "locale_canonicalizer/aliases@1",
+    "locale_canonicalizer/likelysubtags@1",
+    "time_zone/formats@1",
+];
+
+fn main() -> eyre::Result<()> {
+    // manually drop to avoid dhat from printing stats at the end
+    let _profiler = ManuallyDrop::new(dhat::Profiler::new_heap());
+
+    let matches = App::new("ICU4X Zero-Copy verifier")
+        .version("0.0.1")
+        .author("The ICU4X Project Developers")
+        .about("Verify that a Postcard blob deserializes in a zero-copy fashion")
+        .arg(
+            Arg::with_name("VERBOSE")
+                .short("v")
+                .long("verbose")
+                .help("Requests verbose output"),
+        )
+        .arg(
+            Arg::with_name("POSTCARD_FILE")
+                .long("postcard-file")
+                .takes_value(true)
+                .help("Path to .postcard file with all of the data."),
+        )
+        .arg(
+            Arg::with_name("CLDR_ROOT")
+                .long("cldr-root")
+                .value_name("PATH")
+                .help("Path to the CLDR JSON root directory.")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("UPROPS_PATH")
+                .long("uprops-path")
+                .value_name("PATH")
+                .help("Path to the uprops directory.")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("INPUT_FROM_TESTDATA")
+                .long("input-from-testdata")
+                .help("Load input data from the icu_testdata project."),
+        )
+        .arg(
+            Arg::with_name("KEYS")
```

Comment:
I added this because I found it useful to be `--keys specifickey -v` to debug things

Reply:
Also add `ALL_KEYS`?

Reply:
There's currently no difference and we can add that as needed I think

Reply:
The difference with `ALL_KEYS` over `TEST_KEYS` is that in line 220 we wouldn't ignore key errors.

Reply:
That means all-keys would always error, datagen doesn't have proper support for this either. I'm only adding features that'll actually be useful.